input/docs/writing-builds/preprocessor-directives/addin.md

```diff
@@ -1,49 +1,128 @@
+:::{.alert .alert-info}
+Addin directives is not available when running with [Cake Frosting].
+Instead addins can be added as NuGet packages to the [Cake Frosting] project.
+:::
+
+The following URI parameters are supported by the addin directive.
+
+# Source
+
+This is not a named parameter, but it is permitted as per the URI definition.
+By default, the provider will attempt to install addins from nuget.org.
+If the package is hosted on another feed the installation source can be overridden.
+
+## Example
+
+Download from myget.org:
+
+```csharp
+#addin nuget:https://myget.org/f/Cake/?package=Cake.Foo
+```
+
+Install addin from a local directory:
+
+```csharp
+#addin nuget:file://localhost/packages/?package=Cake.Foo
+```
+
+# Package
+
+The name of the NuGet package that should be installed.
+
+## Example
 
 ```csharp
 #addin nuget:?package=Cake.Foo
+```
+
+# Version
+
+The specific version of the NuGet package that should be installed.
+If not provided, the latest version package that is available will be installed.
+
+## Example
+
+```csharp
 #addin nuget:?package=Cake.Foo&version=1.2.3
+```
+
+# Prerelease
+
+To install prerelease tools without defining a specific `version` (`#addin nuget:?package=Cake.Foo&version=1.2.3-beta`), the `prerelease` parameter needs to be passed.
+
+## Example
+
+```csharp
 #addin nuget:?package=Cake.Foo&prerelease
-#addin nuget:https://myget.org/f/Cake/?package=Cake.Foo&prerelease
-// Local feed
-#addin nuget:file://localhost/packages/?package=Cake.Foo
-#addin nuget:file://localhost/packages/?package=Cake.Foo&version=1.2.3
-#addin nuget:file://localhost/packages/?package=Cake.Foo&prerelease
 ```
 
-## Dependencies
+# Dependencies
 
 :::{.alert .alert-success}
 Available since Cake 0.22.0.
 :::
 
-There's an option to fetch and load NuGet dependencies
+By default dependencies defined on the NuGet package are not loaded.
+The `loaddependencies` parameter allows to fetch and load dependent NuGet packages.
+
+:::{.alert .alert-warning}
+This feature requires Cake to be [configured to use in-process NuGet client] instead of external `nuget.exe`.
+:::
+
+## Example
 
 ```csharp
-#addin nuget:?package=foo&loaddependencies=true
-or
-#addin nuget:?package=foo.bar&loaddependencies=false
+#addin nuget:?package=Cake.Foo&loaddependencies=true
 ```
 
-:::{.alert .alert-warning}
-This feature requires Cake to be [configured to use in-process NuGet client](/docs/running-builds/configuration/default-configuration-values#in-process-nuget-installation)
-instead of external `nuget.exe`.
-:::
+# Include
 
-## Include and Exclude Options:
+The `include` parameter allows to define the files which should be included.
+
+## Example
 
 ```csharp
 #addin nuget:?package=Cake.Foo&include=/**/NoFoo.dll
-or
-#addin nuget:?package=Cake.Foo&exclude=/**/Foo.dll
 ```
+
+# Exclude
+
+The `exclude` parameter allows exclude specific files.
+
+## Example
+
+```csharp
+#tool nuget:?package=Cake.Foo&exclude=/**/Foo.dll
+```
+
```

Comment:
Is it worth mentioning somewhere that quotes can be used and are optional?

```cake
#addin nuget:?package=Cake.Foo
```
vs

```cake
#addin "nuget:?package=Cake.Foo"
```